elastic/kibana
# [Index Management] Fix small edit lifecycle bugs

Closes https://github.com/elastic/kibana/issues/168495 https://github.com/elastic/kibana/issues/166385 https://github.com/elastic/kibana/issues/168494 https://github.com/elastic/kibana/issues/168021

## Summary

This pr fixes a few UX inconsistencies regarding lifecycle.enabled, and highlights the data retention state in table and flyout. It also patches a few smaller bugs and unskips a broken api integration test due to an API upgrade.

* if lifecycle.enabled is false, show a `disabled` text to the data retention column and also in the flyout
* if lifecycle.enabled is true and no data_retention value is present, we would want to show a `infinite data retention` text to the data retention column and also in the flyout

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Navigate to console and execute the snippet for creating a test data stream
3. Navigate to index management and click on data streams tab. There are also few other commands for disabling lifecycle, or setting the data_retention to a value not supported by the UI.
4. Verify that:
  * The data stream `b-data-stream` has a time unit of 30nanos and when opening the flyout it also shows data retention 30nanos.
  * When editing the data retention for that DS, it shows 30nanos but the time unit picker doesnt show nano, micro, mili options.
  * When the data stream has `lifecycle.enabled: false`, the UI should show in the table and flyout that the DS has data retention disabled. And the edit modal should show the infinite retention not enabled and an empty form.
  * When the data stream has `lifecycle.enabled: true` but has no `data_retention` configured, then it should show infinite data retention in the table, flyout and edit data retention modal.

<details>
<summary>Create test data stream</summary>

```
PUT _index_template/b-index-template
{
  "index_patterns": ["b-data-stream*"],
  "data_stream": { },
  "priority": 500,
  "template": {
    "lifecycle": {
      "enabled": false
    }
  },
  "_meta": {
    "description": "Template with data stream lifecycle"
  }
}

PUT _data_stream/b-data-stream

PUT b-data-stream/_bulk
{ "create":{ } }
{ "@timestamp": "2099-05-06T16:21:15.000Z", "message": "192.0.2.42 - - [06/May/2099:16:21:15 +0000] \"GET /images/bg.jpg HTTP/1.0\" 200 24736" }
{ "create":{ } }
{ "@timestamp": "2099-05-06T16:25:42.000Z", "message": "192.0.2.255 - - [06/May/2099:16:25:42 +0000] \"GET /favicon.ico HTTP/1.0\" 200 3638" }
```
</details>

<details>
<summary>Set data retention time to 30nanos</summary>

```
PUT _data_stream/b-data-stream/_lifecycle 
{
  "data_retention": "30nanos" 
}
```
</details>

<details>
<summary>Disable lifecycle</summary>

```
PUT _data_stream/b-data-stream/_lifecycle 
{
  "enabled": false
}
```
</details>

<details>
<summary>Screenshots update data retention time units</summary>

![Screenshot 2023-10-11 at 09 29 49](https://github.com/elastic/kibana/assets/1191206/16972b35-d25e-41dc-b783-13aaac176cf7)
![Screenshot 2023-10-11 at 09 29 53](https://github.com/elastic/kibana/assets/1191206/505452d5-c1ef-4fc6-9ae9-f0c900ba7ba0)


</details>
